improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,578 @@
+
+        # 2) Interpret model and blend information from cube attributes
+        self.check_attributes(cube.attributes)
+
+        # 3) Check whether expected coordinates are present
+        coords = get_coord_names(cube)
+        if "spot_index" in coords:
+            self.check_spot_data(cube, coords)
+
+        if self.field_type == self.ANCIL:
+            # there is no clear standard for time coordinates on static ancillaries
+            pass
+        elif (
+            cube.coords("time")
+            and len(cube.coord_dims("time")) == 2
+            and not self.blended
+        ):
+            # 2D time coordinates are only present on global day-max diagnostics that
+            # use a local time zone coordinate. These do not have a 2D forecast period.
+            expected_coords = set(LOCAL_TIME_COORDS + UNBLENDED_TIME_COORDS)
+            expected_coords.discard("forecast_period")
+            self._check_coords_present(coords, expected_coords)
+        elif self.blended:
+            self._check_coords_present(coords, BLENDED_TIME_COORDS)
+        else:
+            self._check_coords_present(coords, UNBLENDED_TIME_COORDS)
+
+        # 4) Check points are equal to upper bounds for bounded time coordinates
+        for coord in ["time", "forecast_period"]:
+            if coord in get_coord_names(cube):
+                self._check_coord_bounds(cube, coord)
+
+        # 5) Check datatypes on data and coordinates
+        try:
+            check_mandatory_standards(cube)
+        except ValueError as cause:
+            self.errors.append(str(cause))
+
+        # 6) Raise collated errors if present
+        if self.errors:
+            raise ValueError("\n".join(self.errors))
+
+
+def display_interpretation(interpreter, verbose=False):
+    """Prints metadata interpretation in human-readable form
+
+    Args:
+        interpreter (MOMetadataInterpreter):
+            Populated instance of MOMetadataInterpreter
+        verbose (bool):
+            Optional flag to include information about the source of the
+            metadata interpretation (eg name, coordinates, attributes, etc)        
+
+    Returns:
+        str:
+            Formatted string describing metadata in human-readable form
+    """
+    if interpreter.unhandled:
+        return f"{interpreter.diagnostic} is not handled by this interpreter"
+
+    def vstring(source_metadata):
+        """Format additional message for verbose output"""
+        return f"    Source: {source_metadata}"
+
+    def format_standard_cases(interpreter):
+        """Format prob / perc / diagnostic information"""
+        field_type = interpreter.field_type.replace("_", " ")
+        diagnostic = interpreter.diagnostic.replace("_", " ")
+        if interpreter.relative_to_threshold:
+            relative_to_threshold = interpreter.relative_to_threshold.replace("_", " ")
+
+        rval = []
+        rtt = (
+            f" {relative_to_threshold} thresholds"
```

Comment:
`relative_to_threshold` is defined inside the `if` on line 525/526. The condition for that `if` is different to the `if` on line 531, so I suspect it's possible to attempt to use a variable which hasn't been defined yet (eg. `interpreter.relative_to_threshold` is False, but `interpreter.field_type == interpreter.PROB` is True). Suggest checking that the logic in these two locations is correct and consistent.

Reply:
The interpreter will raise an error if it's identified a file as being of probability type, which doesn't have a valid relative_to_threshold attribute on the threshold coordinate.  So there is no case in which `interpreter.relative_to_threshold` would be `None` for a probability file at this point.  The display function should never be run on an invalid interpretation (ie one that has raised an error).

Reply:
Update: the interpreter will NOW raise a slightly more helpful error if this attribute is missing!  (Would have raised a `KeyError` around line 217, I have now caught this).